patroni/ha.py

```diff
@@ -554,14 +554,16 @@ def update_cluster_history(self):
         elif not cluster_history or cluster_history[-1][0] != master_timeline - 1 or len(cluster_history[-1]) != 4:
             cluster_history = {line[0]: line for line in cluster_history or []}
             history = self.state_handler.get_history(master_timeline)
+            history[-1].append(self.cluster.leader.name)
```

Comment:
The update_cluster_history() method is executed only by the leader.
```suggestion
            history[-1].append(self.state_handler.name)
```

Reply:
I will get the leader operation and put it in the function patroni/postrgresql/init/get_history